src/ja/server/database/types/work_machine.py

```diff
@@ -46,6 +50,18 @@ def allocate(self,
         @return True if the allocation or the test was successful, False
           otherwise.
         """
+        diff: ResourceAllocation = self._total_resources - allocation
```

Comment:
Regarding code style:
In my PR I added leading underscores to local references in methods.
Should we add underscores to local references @ammen99 @nikolatzotchev @M1keReck @jooe1 ?

Reply:
> _single_leading_underscore: weak "internal use" indicator. E.g. from M import * does not import objects whose names start with an underscore.
That is from pep8, I'm fine with both if someone else has some preferences.

Reply:
Yes we should, leading underscores indicate that a reference is private so that's actually necessary.

Reply:
>Yes we should, leading underscores indicate that a reference is private so that's actually necessary.

Well, was there perhaps a misunderstanding?
What I meant is that I put underscores in front of all local references, not just object attributes.
So in this case it would be `_diff` instead of `diff`.

Reply:
There was, we don't need leading underscores for references inside a method because they can't be accessed outside of that method. Object attributes can be accessed outside of their class so an underscore is necessary.